providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/grpc/EventStreamObserver.java

```diff
@@ -52,37 +57,38 @@ public void onNext(EventStreamResponse value) {
                 this.handleProviderReadyEvent();
                 break;
             default:
-                log.debug("unhandled event type {}", value.getType());
+                log.debug("Unhandled event type {}", value.getType());
         }
     }
 
+    /**
+     * Called when an error occurs in the stream.
+     *
+     * @param throwable the error that occurred
+     */
     @Override
     public void onError(Throwable throwable) {
-        if (Boolean.TRUE.equals(shouldRetrySilently.get())) {
-            log.debug("Event stream error, trying to recover", throwable);
-        } else {
-            log.error("Event stream error", throwable);
-            if (this.cache.getEnabled()) {
-                this.cache.clear();
-            }
-            this.onConnectionEvent.accept(false, Collections.emptyList());
+        if (this.cache.getEnabled().equals(Boolean.TRUE)) {
```

Comment:
Another nit, but generally I prefer such expressions in the reverse order in Java, since `Boolean.TRUE` will obviously never ever be `null`, so it's less likely to cause NPEs whereas if `this.cache.getEnabled()` becomes nullable at some point (maybe it's a boxed type) you can see NPEs here calling `.equals`.

Reply:
Yeah, i actually just followed the sonar linter suggestion here